### PR TITLE
Adding coveralls

### LINKED ISSUE #165: Checking test coverage
In order to keep this repo going well, another thing we should look into is test coverage.   
Now [CodeClimate](https://docs.codeclimate.com/docs/setting-up-test-coverage) does offer code coverage testing, but it might not be right for us. 
My main concern _against_ using CodeClimate for test coverage is that it is not multi-language friendly. 

But it _is_ something we have already integrated with out project so it would be the easiest to setup. Thoughts?

### PR DESCRIPTION
**Pull request by isaac-gs**
## Description
I added coveralls to the requirements.txt file and added sending results as an after_success stage for Travis CI.

## Related Issue(s)
Closes #165 

## Motivation and Context
We needed the brutal honest truth of how bad our test coverage. We need the pain of reality.

## How Has This Been Tested?
I ran Travis CI. It builds.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (documentation, refactoring, basically not a feature or a bug fix)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by asclines** [comment]
So it's the libraries that make it drop down to 25%?

**Comment by isaac-gs** [comment]
@asclines yup

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[run]
+include = */modules/*
+omit =
```
Comment (asclines) [inline .coveragerc:3]: So it ignores our test files?

File: .coveragerc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[run]
+include = */modules/*
```
Comment (asclines) [inline .coveragerc:2]: So this makes it _only_ checkout our modules? Like a whitelist?

File: .travis.yml [hunk 3]
```diff
@@ -9,4 +9,6 @@ before_script:
 - python tests/unittest_data/test_file_downloader.py
 # command to run tests
 script:
-- python -m unittest discover
+- coverage run -m unittest discover
```
Comment (asclines) [inline .travis.yml:12]: So coverage runs our tests _in addition to_ checking the coverage?

**Review by asclines — APPROVED** [review]

File: .coveragerc (hunk 2 above)
Comment (isaac-gs) [reply .coveragerc:2]: Yup

File: .coveragerc (hunk 1 above)
Comment (isaac-gs) [reply .coveragerc:3]: Yeah, we don't want test coverage on test files

File: .travis.yml (hunk 3 above)
Comment (isaac-gs) [reply .travis.yml:12]: Yup

**Pull request merged**